### PR TITLE
Feature: add pga losses and losses submodule

### PR DESCRIPTION
**Pull request by 0xangelo**
Signed-off-by: Ângelo Lovatto <angelolovatto@gmail.com>

### CODE REVIEW

File: mapo/agents/mapo/losses.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        batch_tensors[SampleBatch.ACTIONS],
+        batch_tensors[SampleBatch.NEXT_OBS],
+    )
+    return -tf.reduce_mean(model.compute_states_log_prob(obs, actions, next_obs))
+
+
+def dynamics_pga_loss(batch_tensors, model, actor_loss, config):
+    """Compute Policy Gradient Aware dynamics loss"""
+    gmapo = tf.gradients(actor_loss, model.actor_variables)
+    flat_gmapo = tf.concat([tf.reshape(grad, [-1]) for grad in gmapo], axis=0)
```
Comment (thiagopbueno) [inline mapo/agents/mapo/losses.py]: Maybe it would be better if we leave this concatenation step as an option to the kernel function. 
Since not all kernels are additive, I'm not sure this is the best way to go. You should be able to test both ways (e.g., the sum of the results of the kernels for each variable Vs. the kernel applied to the concatenated vector all at once) ... 

So, I think it would make a more flexible solution to pass to the kernel both lists of gradient vectors (aligned wrt to the actor variables) and let the kernel compute the scalar loss... ok?

File: mapo/agents/mapo/tests/test_losses.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    return request.param
+
+
+@pytest.fixture
+def kernel_fn():
+    return lambda x, y: tf.norm(x - y, axis=0)
+
+
+# skip true dynamics test until env dynamics log_prob is done
+@pytest.fixture(params=[False, pytest.param(True, marks=pytest.mark.skip)])
```
Comment (thiagopbueno) [inline mapo/agents/mapo/tests/test_losses.py]: Will code `env.compute_log_prob` to unskip that just after this review...

**Review by thiagopbueno — CHANGES_REQUESTED** [review]

File: mapo/agents/mapo/losses.py (hunk 1 above)
Comment (0xangelo) [reply mapo/agents/mapo/losses.py]: Makes sense, I'll implement it right away.

File: mapo/agents/mapo/tests/test_losses.py (hunk 2 above)
Comment (thiagopbueno) [reply mapo/agents/mapo/tests/test_losses.py]: Done! If you accept the feature/env-log-prob PR I've just opened you might be able to run these tests...

**Review by thiagopbueno — CHANGES_REQUESTED** [review]

**Review by thiagopbueno — APPROVED** [review]
LGTM!
